### PR TITLE
End to End TLS SSL step #3

### PR DESCRIPTION
**Pull request by mnriem**

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java [hunk 1]
```diff
@@ -83,13 +82,13 @@ public KeyVaultKeyStore() {
         String tenantId = System.getProperty("azure.keyvault.tenantId");
         String clientId = System.getProperty("azure.keyvault.clientId");
         String clientSecret = System.getProperty("azure.keyvault.clientSecret");
-        keyVault = new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret);
+        keyVaultClient = new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret);
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java:85]:
```suggestion
        keyVaultClient = new KeyVaultClient(keyVaultUrl, tenantId, clientId, clientSecret);
```
All occurrences, really :)

**Review by vcolin7 — COMMENTED** [review]
Most things look good to me. I think it would be very helpful if these PRs had a brief description of what changes were made to look out for them, especially when there's [another unmerged PR](https://github.com/Azure/azure-sdk-for-java/pull/16708) this is based on.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by chenrujun** [comment]
Hi, @mnriem , 

I merged this PR.
Please create another PR to solve the unresolved conversations in the last 3 PRs:

 - https://github.com/Azure/azure-sdk-for-java/pull/16452
 - https://github.com/Azure/azure-sdk-for-java/pull/16708
 - https://github.com/Azure/azure-sdk-for-java/pull/16773

cc: @vcolin7 , FYI.

**Pull request merged**